tests/integration/handling-api-messages.js

```diff
@@ -0,0 +1,20 @@
+describe("handling api messages", function () {
+
+  it("handles and re-hydrates commands", function () {
```

Comment:
Isn't the construction of the command where the only serialisation taking place, ie not in Space.messaging.Api and out of the SUT? (we could easily just drop in a factory-built test command in this case).

The current implementation we would have these scenarios to test here:

- `sends a sent command on the command bus if passes validation`
- `does not send a sent command on the command bus if fails validation`
-  `publishes a sent event on the event bus if it passes validation`
-  `does not publish a sent event on the event bus if it passes validation`

I'd like to see us take an opinionated approach using CQRS principles here, only supporting commands in `.send()`. With the above tests we're exposing this confusing interface of ['sending' events to the API](https://github.com/meteor-space/messaging/blob/develop/source/api.coffee#L23)) which currently is a request mechanism since it has the chance to reject. If we want events to propagate from the client to the server we need to only allow it via a simple mapping, with the interface of `Space.messaging.Api.publish()`

With those changes the test case becomes:

For Space.messaging.Api.send()
- `sends a handled command on the command bus if passes validation`
- `does not send a handled command on the command bus if fails validation`

For Space.messaging.Api.publish()
-  `publishes a subscribed event on the event bus'
-  `does not publish a subscribed event on the event bus if it passes validation`